model/arithmetic.sail

```diff
@@ -6,6 +6,16 @@
 /*  SPDX-License-Identifier: BSD-2-Clause                                                */
 /*=======================================================================================*/
 
+/* Reverse bits in each byte */
+val brev8 : forall 'm, 'm >= 0. (bits('m)) -> bits('m)
+function brev8(input) = {
+  assert('m % 8 == 0, "input length must be a multiple of 8");
```

Comment:
You could enforce this in the type system:

```
function brev8 forall 'm, 'm >= 0 & mod('m, 8) == 0 . (input : bits('m)) -> bits('m) = {
  var output : bits('m) = zeros();
  foreach (i from 0 to ('m - 8) by 8)
    output[i+7..i] = reverse(input[i+7..i]);
  output
}
```

Looks like `get_velem()` can return a multiple of 4 bits though?